config/main.py

```diff
@@ -1875,6 +1875,33 @@ def override_config_by(golden_config_path):
     return
 
 
+def generate_sysinfo(config_input, ns=None):
+    # Generate required sysinfo for Golden Config.
+    device_metadata = config_input.get('DEVICE_METADATA')
+
+    if not device_metadata or 'localhost' not in device_metadata:
+        return
+
+    if ns:
+        asic_role = device_metadata.get('localhost', {}).get('sub_role')
+        switch_type = device_metadata.get('localhost', {}).get('switch_type')
+
+        if ((switch_type is not None and switch_type.lower() == "chassis-packet") or
+                (asic_role is not None and asic_role.lower() == "backend")):
+            mac = device_info.get_system_mac(namespace=ns)
+        else:
+            mac = device_info.get_system_mac()
+    else:
+        mac = device_info.get_system_mac()
```

Comment:
I notice code duplication of https://github.com/sonic-net/sonic-buildimage/blob/d74055e12ca0ac4f44342e74ea097f03720663f7/src/sonic-config-engine/sonic-cfggen#L382-L388.
Possible to reuse code?

Reply:
Could you remove code dup in next PR?